build/quarks-secret/_vendir/deploy/helm/quarks-secret/values.yaml

```diff
@@ -0,0 +1,55 @@
+## Default values for Quarks Secret Helm Chart.
+## This is a YAML-formatted file.
+## Declare variables to be passed into your templates.
+
+
+# applyCRD is a boolean to control the installation of CRD's.
+applyCRD: true
```

Comment:
We would like to add a yaml file into the helm chart itself (or into the build dir if we must) for the CRD resource which is also conditionally rendered when we set applyCRD: false because
- We prefer to have least privilege for the operator
- We would like kapp to own all of the resources we are deploying, so it needs to include the explicit CRD yaml in its underlying kubectl apply
- It's easier to manage the creation of QuarksSecrets themselves if we are also creating the CRD on which they are dependent (we can order these in kapp, for instance)

As a side note it was a bit weird that the authoritative CRD config is inlined json in the go code for the operator as it makes it harder for consumers to find. We ended up taking the result of the operator from our own kubernetes cluster to produce a standalone yaml file for the CRD.

That file is included below as txt, but is yaml formatted.
[quarkssecret-crd.txt](https://github.com/cloudfoundry/cf-for-k8s/files/4921948/quarkssecret-crd.txt)

Reply:
There is an (outdated?) dump of the CRD in docs/crds, we had them in the file system originally, but having the CRD in code made CI and such a lot easier.

Reply:
Helm's `crds` folder will not be taken into account for updates. I suppose this is not relevant here as the build script only uses `helm template`, but it would not work for regular `helm update - i` based workflows.

Reply:
> but having the CRD in code made CI and such a lot easier.

Makes sense. I imagine it would have been a lot of boilerplate to update if it had to live outside of code.

> Helm's crds folder will not be taken into account for updates.
Oh I didn't know Helm had conventions around crd resources, I'll do some reading. Thanks for the reminder that while cf-for-k8s is the current context of the integration, the quarks secret repo itself has a desire to properly support the helm chart and we should keep that tension in mind when making requests to update the chart.

Reply:
pulled in a new chart where this issue is resolved.